client/boxoptions/options.go

```diff
@@ -77,57 +84,109 @@ var boxOptions = option.Options{
 			Tag:  "dns-out",
 			// Options: &option.DNSOptions{},
 		},
+		{
+			Type: constant.TypeOutline,
+			Tag:  "outline-out",
+			Options: &option.OutboundOutlineOptions{
+				DNSResolvers: []option.DNSEntryConfig{
+					{
+						TLS: &option.TLSEntryConfig{
+							Name:    "dns.google",
+							Address: "8.8.8.8:853",
+						},
+					},
+					{
+						HTTPS: &option.HTTPSEntryConfig{
+							Name: "dns.google",
+						},
+					},
+					{
+						HTTPS: &option.HTTPSEntryConfig{
+							Name:    "cloudflare-dns.com.",
+							Address: "cloudflare.net.",
+						},
+					},
+					{
+						HTTPS: &option.HTTPSEntryConfig{
+							Name:    "doh.dns.sb",
+							Address: "https://doh.dns.sb/dns-query",
+						},
+					},
+					{
+						System: &struct{}{},
+					},
+				},
+				TLS:         []string{"", "split:1", "split:2,20*5", "split:200|disorder:1", "tlsfrag:1"},
```

Comment:
`split` and `disorder` are mutations at the TCP level, not the TLS level.  But it seems like those things are now in our sing-box fork, so maybe that's slightly out of scope of this PR.  Just noting that this is a little odd 🤷

Reply:
This is how the stream dialer receive those techniques from the config

Reply:
The [TLS name](https://github.com/Jigsaw-Code/outline-sdk/blob/main/x/smart/stream_dialer.go#L99) in this case is just how they receive the techniques and it's not exclusive to TLS

Reply:
But I agree it's weird and we could probably update and rename at the sing-box implementation

Reply:
Right -- the idea is that these are the techniques it will use for TLS traffic -- i.e. for https. It doesn't have support for doing anything for arbitrary TCP traffic.

Reply:
> The [TLS name](https://github.com/Jigsaw-Code/outline-sdk/blob/main/x/smart/stream_dialer.go#L99) in this case is just how they receive the techniques and it's not exclusive to TLS

Pretty sure it is exclusive to TLS -- I don't think the smart dialer does anything with other types of traffic.

Reply:
> I don't think the smart dialer does anything with other types of traffic.

Yes, after [looking through the code](https://github.com/Jigsaw-Code/outline-sdk/blob/ac9482090891ae27986d7d5e83408f3231471f1e/x/smart/stream_dialer.go#L244-L272) it searches for the TLS dialer that works with those configurations in parallel (by dialing and handshaking) and use the selected dialer with the technique for stream dial

Reply:
Looking again through the code it calls it a `tlsDialer` but it could be a TCP dialer wrapped by the split implementation returned from the `config.NewStreamDialer`

Reply:
So it dials to the test domain with the stream dialer that not necessarily is a TLS dialer, wraps the connection as a TLS, do the handshake and the selected dialer is returned

Reply:
We could probably rename the `TLS` property to proxyless techniques or something else that makes sense because it's not really related to TLS but that would be different then the smart dialer config from outline. Should we still change that?

Reply:
hmmmm ok now I've debugged locally and whatever dialer being created from the given config, it's using the `transport.HappyEyeballsStreamDialer` as a inner dialer (that uses the outbound dialer created at the sing-box PR)

Reply:
Oh, interesting!  Odd it restricts TCP-level stuff to TLS, but I guess their use case is narrow.

Sorry for the distraction!